### PR TITLE
Presets fix: proper Esc key handling for preset dialogs

### PR DESCRIPTION
**Pull request by limonspb**
By default the ESC key closes `<dialog>s` that were opened with showModal()
For this reason, the dialog-clear logic can't be inside the "Close" button click. It has to be inside "close" event. That will make the behavior consistent regardless if the user clicked "close" or pressed the ESC key.

This PR also fixes spelling (reinitiali**s**eConnection -> reinitiali**z**eConnection) leftover after this PR: https://github.com/betaflight/betaflight-configurator/pull/2758

After this PR the following dialogs must behave well when user presses the ESC key inside of them:
Preset detailed dialog
Dialog with preset source selection
Dialog with CLI errors on load backup/applying presets.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by blckmn — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by chmelevskij — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> PASS
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Pull request merged**
